Tests/PackageLoadingTests/PD4_2LoadingTests.swift

```diff
@@ -499,24 +499,39 @@ class PackageDescription4_2LoadingTests: PackageDescriptionLoadingTests {
         }
     }
 
-    func testURLContainsNotAbsolutePath() throws {
-        let stream = BufferedOutputByteStream()
-        stream <<< """
-        import PackageDescription
-        let package = Package(
-            name: "Trivial",
-            dependencies: [
-                .package(url: "file://../best", from: "1.0.0"),
-            ],
-            targets: [
-                .target(
-                    name: "foo",
-                    dependencies: []),
-            ]
-        )
-        """
-
-        XCTAssertManifestLoadThrows(ManifestParseError.invalidManifestFormat("file:// URLs cannot be relative, did you mean to use `.package(path:)`?", diagnosticFile: nil), stream.bytes)
+    func testFileURLsWithHostnames() throws {
+        enum ExpectedError {
+          case relativePath
+          case unsupportedHostname
+        }
+        let urls: [(String, ExpectedError)] = [
+          ("file://../best", .relativePath), // Possible attempt at a relative path.
+          ("file://somehost/bar", .unsupportedHostname), // Obviously non-local.
+          ("file://localhost/bar", .unsupportedHostname), // Local but non-trivial (e.g. on Windows, this is a UNC path).
+        ]
+        for (url, expectedError) in urls {
+            let stream = BufferedOutputByteStream()
+            stream <<< """
+            import PackageDescription
+            let package = Package(
+                name: "Trivial",
+                dependencies: [
+                    .package(url: "\(url)", from: "1.0.0"),
+                ],
+                targets: [
+                    .target(
+                        name: "foo",
+                        dependencies: []),
+                ]
+            )
+            """
+            switch expectedError {
+            case .relativePath:
+              XCTAssertManifestLoadThrows(ManifestParseError.invalidManifestFormat("file:// URLs cannot be relative, did you mean to use `.package(path:)`?", diagnosticFile: nil), stream.bytes)
```

Comment:
same nitpick as the previous comment:

```suggestion
              XCTAssertManifestLoadThrows(ManifestParseError.invalidManifestFormat("file:// URLs cannot be relative, did you mean to use '.package(path:)'?", diagnosticFile: nil), stream.bytes)
```